packages/editor/src/components/browser/function/FunctionBrowser.tsx

```diff
@@ -123,19 +159,34 @@ const FunctionBrowser = (props: { value: string; onChange: (value: BrowserValue)
           value: globalFilter,
           onChange: newFilterValue => {
             setGlobalFilter(newFilterValue);
-            setExpanded(true);
+            setExpanded(newFilterValue.length > 0 ? true : { [0]: true });
+            setRowSelection({});
+            setRowNumber(100);
           }
         }}
+        ref={parentRef}
       >
         <tbody>
-          {table.getRowModel().rows.map(row => (
-            <SelectRow key={row.id} row={row} onDoubleClick={props.onDoubleClick}>
-              {row.getVisibleCells().map(cell => (
-                <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
-              ))}
-            </SelectRow>
-          ))}
+          {virtualizer.getVirtualItems().map(virtualRow => {
+            const row = rows[virtualRow.index] as Row<Function>;
+            return (
+              <SelectRow key={row.id} row={row} onDoubleClick={props.onDoubleClick}>
+                {row.getVisibleCells().map(cell => (
+                  <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
+                ))}
+              </SelectRow>
+            );
+          })}
         </tbody>
+        {rowNumber < rows.length && (
+          <TableFooter>
+            <TableShowMore
+              colSpan={4}
+              showMore={() => setRowNumber(rowNumber + 100)}
```

Comment:
normally if you use the old value to calculate the new one the method `setRowNumber(old => old + 100);` fits better.